code/modules/client/preferences.dm

```diff
@@ -966,12 +966,15 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 					var/result = input(user, "Select a species", "Species Selection") as null|anything in GLOB.roundstart_races
 
 					if(result)
-						var/newtype = GLOB.species_list[result]
-						pref_species = new newtype()
-						//Now that we changed our species, we must verify that the mutant colour is still allowed.
-						var/temp_hsv = RGBtoHSV(features["mcolor"])
-						if(features["mcolor"] == "#000" || (!(MUTCOLORS_PARTSONLY in pref_species.species_traits) && ReadHSV(temp_hsv)[3] < ReadHSV("#7F7F7F")[3]))
-							features["mcolor"] = pref_species.default_color
+						var/datum/species/newtype = GLOB.species_list[result]
+						if(newtype.is_player_eligible(parent))

```

Comment:
this won't work, `newtype` is going to be a path, gotta create it first

Reply:
newtypes are being added too? Gundamstation13 when